activerecord/lib/active_record/connection_adapters/mysql/schema_definitions.rb

```diff
@@ -50,6 +50,31 @@ module ColumnMethods
         deprecate :unsigned_float, :unsigned_decimal, deprecator: ActiveRecord.deprecator
       end
 
+      # = Active Record MySQL Adapter \Index Definition
+      class IndexDefinition < ActiveRecord::ConnectionAdapters::IndexDefinition
+        attr_reader :enabled
+
+        def initialize(*args, **kwargs)
+          @enabled = kwargs.key?(:enabled) ? kwargs.delete(:enabled) : true
+          super
+        end
+
+        def enabled=(value)
+          return if value.nil?
```

Comment:
IMO we should make this a normal accessor and guard the assignment call site instead. Seems like just one place, right?